lib/LibManifestPlugin.js

```diff
@@ -104,7 +104,11 @@ class LibManifestPlugin {
 							)
 								.filter(Boolean)
 								.reduce((obj, item) => {
-									obj[item.ident] = item.data;
+									if (item.ident === item.data.id) {
```

Comment:
The logic should be placed in DllReferencePlugin instead. It already handles `extensions`. You should add a option `mainFiles: ["index"]` which is resolved in addition.